Tests/NFUnitTestArithmetic/UnitTestFormat.cs

```diff
@@ -38,34 +38,44 @@ public enum Case
         }
 
         [TestMethod]
-        public void StringFormat()
+        public void StringFormat_00()
         {
             // Test a null string in String.Format - should be treated like an empty string.
             string nullArg = null;
             Assert.Equal(string.Format("Value is {0}", nullArg), "Value is ", "");
+        }
 
+        [TestMethod]
+        public void StringFormat_01()
+        {
             // catch an exception if the format string is null
             string nullFormat = null;
             Assert.Throws(typeof(NullReferenceException), () => { string.Format(nullFormat, 12345.67); });
+        }
 
+        [TestMethod]
```

Comment:
You can remove [TestMethod] parameter when using [DataRow] :)

Reply:
🤔  maybe I could... but I don't think we should... Check [here](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-with-mstest#add-more-features) please.

The official usage guidance calls for both. I get it that we may handle this differently but then we have to take care of it internally so it doesn't break the official usage.

Maybe process DataRow before TestMethod and then don't call it "a second time" or make a mutually exclusive check... not sure what would be more efficient.

Reply:
With current implementation of DataRow - it will probably fail. Test framework will try to invoke method and pass "null" as parameters.

" mutually exclusive check" - do you mean something like this:
if (testMethod.Contains(DataRowAttribute) && test.Method.Contains(TestMethodAttrubite)
{
throw Exception();
}

Or just ignore TestMethod attribute when there is DataRow?

Reply:
I would go with the latter. Test will be executed (as intended) just that with the correct context (using DataRow(s)). And the "doomed to fail" execution of TestMethod is skipped.